framework/src/main/java/org/checkerframework/common/returnsreceiver/FluentAPIGenerator.java

```diff
@@ -0,0 +1,136 @@
+package org.checkerframework.common.returnsreceiver;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.BugInCF;
+import org.checkerframework.javacutil.TypesUtils;
+
+/**
+ * Wrapper class for {@link FluentAPIGenerators} Enum to keep it private. The only use of the enum
+ * is to support fluent API generators so the checker can add @This annotations on method return
+ * types when these generators has been used. To check whether a method is created by any of the
+ * generators defined in the {@link FluentAPIGenerators} Enum and returns {@code this}, simply call
+ * the {@link FluentAPIGenerator#check} on the annotated type of the method signature.
+ */
+public class FluentAPIGenerator {
+
+    /**
+     * This is the only public method of this class
```

Comment:
I'm not sure that's the best description of the method you could come up with...